include/crm/pengine/print_ml_status.h

```diff
@@ -0,0 +1,128 @@
+    mon_output_plain,
+    mon_output_console,
+    mon_output_xml,
+    mon_output_html,
+    mon_output_cgi
+} output_format = mon_output_console;
+
+struct print_params_t
+{
+    unsigned int show; // = mon_show_default;
+    gboolean group_by_node; // = FALSE;
+    gboolean inactive_resources; // = FALSE;
+    gboolean print_timing; // = FALSE;
+    gboolean fence_history; // = FALSE
+    gboolean fence_full_history; // = FALSE
+    gboolean print_brief; // = FALSE;
+    gboolean print_pending; // = TRUE;
+    gboolean print_clone_detail; // = FALSE;
+    /* FIXME allow, detect, and correctly interpret glob pattern or regex? */
+    const char *print_neg_location_prefix; // = "";
+};
+
+/* Define exit codes for monitoring-compatible output
+ * For nagios plugins, the possibilities are
+ * OK=0, WARN=1, CRIT=2, and UNKNOWN=3
+ */
+#define MON_STATUS_WARN    CRM_EX_ERROR
+#define MON_STATUS_CRIT    CRM_EX_INVALID_PARAM
+#define MON_STATUS_UNKNOWN CRM_EX_UNIMPLEMENT_FEATURE
+
+/* Convenience macro for prettifying output (e.g. "node" vs "nodes") */
+#define s_if_plural(i) (((i) == 1)? "" : "s")
+
+#if CURSES_ENABLED
+#  define print_dot() if (output_format == mon_output_console) { \
+        printw(".");                            \
+        clrtoeol();                             \
+        refresh();                              \
+    } else {                                    \
+        fprintf(stdout, ".");                   \
+    }
+#else
+#  define print_dot() fprintf(stdout, ".");
+#endif
+
+#if CURSES_ENABLED
+#  define print_as(fmt, args...) if (output_format == mon_output_console) { \
+        printw(fmt, ##args);                            \
+        clrtoeol();                                     \
+        refresh();                                      \
+    } else {                                            \
+        fprintf(stdout, fmt, ##args);                   \
+    }
+#else
+#  define print_as(fmt, args...) fprintf(stdout, fmt, ##args);
+#endif
+
+int get_resource_display_options(void);
+void print_cluster_summary(FILE *stream, pe_working_set_t *data_set);
+int count_resources(pe_working_set_t * data_set, pe_resource_t * rsc);
+void print_resources(FILE *stream, pe_working_set_t *data_set, int print_opts);
+void print_node_attributes(FILE *stream, pe_working_set_t *data_set);
+char *get_node_display_name(pe_node_t *node);
+void print_node_summary(FILE *stream, pe_working_set_t * data_set, gboolean operations);
+void print_failed_actions(FILE *stream, pe_working_set_t *data_set);
+void print_stonith_history(FILE *stream, stonith_history_t *history);
+void print_stonith_action(FILE *stream, stonith_history_t *event);
+void print_cluster_tickets(FILE *stream, pe_working_set_t * data_set);
+void print_neg_locations(FILE *stream, pe_working_set_t *data_set);
+void print_xml_status(FILE *stream, pe_working_set_t *data_set, stonith_history_t *stonith_history
+		      , struct print_params_t print_params);
```

Comment:
Would it make sense to support printing of other formats such as print_status() and print_html_status() also from here? Or wrapper them with a common print_status()  function supporting parameter ```output_format```?

Reply:
This is a good idea. I could create a next commit within this pull request for it. Please acknowledge.

Reply:
And we will need to rename the existing print_status into print_plain_status functions